src/common/diagnostics.py

```diff
@@ -57,7 +58,7 @@ def setup_diagnostics_variables(model, start_time, options, solver_options):
                                     solver_name=="ExplicitEuler"
                                     )
 
-        states_list = model.get_states_list() if isinstance(model, FMUModelME2) else []
+        states_list = model.get_states_list() if isinstance(model, (FMUModelME2, FMUModelME3)) else []
```

Comment:
Would it be possible to change this to `hasattr(model, get_states_list)`? With the purpose of making it more scalable and dont need updates int he future with additional classes.